From cec97e1e8e0dc407ea13b583ca6b15dce98b0fbd Mon Sep 17 00:00:00 2001
From: LuK1337 <priv.luk@gmail.com>
Date: Sat, 10 Apr 2021 10:55:36 +0200
Subject: [PATCH] Allow enabling support for audio and debug accessories

This effectively allows us to get rid of analog device
unsupported notification on devices that can do analog
audio just fine. I doubt that anyone will actually need
to toggle debug accessory but it doesn't hurt to support
it too, does it?

Change-Id: I09b107726e9ba07c940732fd3c628701b7aaec30
---
 hal/Android.bp | 5 ++++-
 hal/Usb.cpp    | 6 ++++++
 2 files changed, 10 insertions(+), 1 deletion(-)

diff --git a/hal/Android.bp b/hal/Android.bp
index a24756d..efa44cf 100644
--- a/hal/Android.bp
+++ b/hal/Android.bp
@@ -18,7 +18,10 @@ cc_defaults {
 
 cc_binary {
     name: "android.hardware.usb@1.2-service-qti",
-    defaults: ["qti_usb_hal_defaults"],
+    defaults: [
+        "qti_usb_hal_defaults",
+        "qti_usb_hal_supported_modes_defaults"
+    ],
     shared_libs: [
         "android.hardware.usb@1.0",
         "android.hardware.usb@1.1",
diff --git a/hal/Usb.cpp b/hal/Usb.cpp
index fc69437..17b0b84 100644
--- a/hal/Usb.cpp
+++ b/hal/Usb.cpp
@@ -492,6 +492,12 @@ Status getPortStatusHelper(hidl_vec<PortStatus> *currentPortStatus_1_2,
         (*currentPortStatus_1_2)[i].status_1_1.status.supportedModes = V1_0::PortMode::DFP;
       } else {
         (*currentPortStatus_1_2)[i].status_1_1.supportedModes = PortMode_1_1::UFP | PortMode_1_1::DFP;
+#ifdef SUPPORTS_AUDIO_ACCESSORY
+        (*currentPortStatus_1_2)[i].status_1_1.supportedModes |= PortMode_1_1::AUDIO_ACCESSORY;
+#endif
+#ifdef SUPPORTS_DEBUG_ACCESSORY
+        (*currentPortStatus_1_2)[i].status_1_1.supportedModes |= PortMode_1_1::DEBUG_ACCESSORY;
+#endif
         (*currentPortStatus_1_2)[i].status_1_1.status.supportedModes = V1_0::PortMode::NONE;
         (*currentPortStatus_1_2)[i].status_1_1.status.currentMode = V1_0::PortMode::NONE;
 
